src/EnergyPlus/TARCOGCommon.cc

```diff
@@ -70,14 +70,15 @@ namespace TARCOGCommon {
     // PURPOSE OF THIS MODULE:
     // A module which contains common TARCOG functions and subroutines
 
-    bool IsShadingLayer(int const layertype)
+    bool IsShadingLayer(const EnergyPlus::TARCOGParams::TARCOGLayerType layertype)
     {
 
         // Using/Aliasing
         using namespace TARCOGParams;
 
-        return layertype == VENETBLIND_HORIZ || layertype == VENETBLIND_VERT || layertype == WOVSHADE || layertype == PERFORATED ||
-               layertype == BSDF || layertype == DIFFSHADE;
+        return layertype == TARCOGLayerType::VENETBLIND_HORIZ || layertype == TARCOGLayerType::VENETBLIND_VERT ||
+               layertype == TARCOGLayerType::WOVSHADE || layertype == TARCOGLayerType::PERFORATED || layertype == TARCOGLayerType::BSDF ||
+               layertype == TARCOGLayerType::DIFFSHADE;
     }
```

Comment:
Perfect spot for the `BITF` trick.

Reply:
I did not understand. Do you mean bit fields?

Reply:
Yes, `BITF_TEST_ANY(BITF(layertype), BITF(TARCOGLayerType::VENETBLIND_HORIZ) | BITF(TARCOGLayerType::VENETBLIND_VERT) | ... )`. One instruction instead of 12.

Reply:
Awesome. Will do!

Reply:
Is it recommended to use `BITF` in other places as well?